pkg/objstore/read_only_file.go

```diff
@@ -33,39 +33,36 @@ func Download(ctx context.Context, name string, src BucketReader, dir string) (*
 	return f, nil
 }
 
-func download(ctx context.Context, name string, src BucketReader, dir string) (f *ReadOnlyFile, err error) {
+func download(ctx context.Context, name string, src BucketReader, dir string) (*ReadOnlyFile, error) {
```

Comment:
I think a linter that catches such cases might be helpful. Naked return is generally considered harmful, but it's unavoidable (or convenient) in some cases